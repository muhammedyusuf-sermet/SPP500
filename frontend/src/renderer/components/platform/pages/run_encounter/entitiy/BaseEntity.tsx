import * as React from 'react';

import 'bulma/css/bulma.css';

import { isDeepStrictEqual } from 'util';
import { ExpansionPanel, ExpansionPanelSummary, ExpansionPanelDetails, Typography, FormControl, InputLabel, Input, Grid, FormControlLabel, Switch, Button } from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

export interface IBaseEntityProps {
	Id: number
	Initiative: number;
	View: (event: React.MouseEvent<HTMLButtonElement>) => void,
	Edit: (event: React.MouseEvent<HTMLButtonElement>) => void,
	// initial values
	Entity: {
		EntityType: keyof EntityTypes;
		Id: number;
		Name: string;
		ArmorClass: number;
		HitPoints: number;
	}
}

export const EntityConditions = [
	'Blinded',
	'Charmed',
	'Deafened',
	'Frightened',
	'Grappled',
	'Incapacitated',
	'Invisible',
	'Paralyzed',
	'Petrified',
	'Poisoned',
	'Prone',
	'Restrained',
	'Stunned',
	'Unconscious',
]

export type EntityTypes = {
	Player: string,
	Monster: string
}

export interface IBaseEntityState {
	[key: string]: boolean | number | undefined;
	// data
	Blinded?: boolean,
	Charmed?: boolean,
	Deafened?: boolean,
	Frightened?: boolean,
	Grappled?: boolean,
	Incapacitated?: boolean,
	Invisible?: boolean,
	Paralyzed?: boolean,
	Petrified?: boolean,
	Poisoned?: boolean,
	Prone?: boolean,
	Restrained?: boolean,
	Stunned?: boolean,
	Unconscious?: boolean,
	CurrentHitPoints?: number;
}

export class BaseEntity extends React.Component<IBaseEntityProps, IBaseEntityState> {
	constructor(props: IBaseEntityProps) {
		super(props);
		this.state = {
			CurrentHitPoints: this.props.Entity.HitPoints
		};
	}

	componentWillReceiveProps(nextProps: IBaseEntityProps) {
		if (isDeepStrictEqual(this.props.Entity, nextProps.Entity) == false)
			this.setState({
				CurrentHitPoints: nextProps.Entity.HitPoints
			});
	}

	stringToNumber = (toConvert : string) => {
		return isNaN(parseInt(toConvert)) ? undefined : parseInt(toConvert);
	}

	handleNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		let value = this.stringToNumber(event.target.value);
		value = value ? value : 0;
		const name = event.currentTarget.name;
		this.setState({
			[name]: value,
		});
	}

	handleConditionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		const value = event.target.value;
		this.setState((prevState) => ({
			[value]: !prevState[value]
		}))
	}

	render() {
		let conditions = '';
		for (let con of EntityConditions)
			if(this.state[con])
				conditions = conditions + ', ' + con;
		conditions = conditions.substring(2);
		return (
			<ExpansionPanel CollapseProps={{timeout: 100}} >
				<ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
					<Grid container spacing={8} >
						<Grid item xs={6}>
							<Typography id='Name' variant='h6' >{this.props.Entity.Name}</Typography>
						</Grid>
						<Grid item xs={6}>
							<Typography id='Initiative' className='body1' >{'Initative: '+this.props.Initiative}</Typography>
						</Grid>
						<Grid item xs={6}>
							<Typography id='ArmorClass' className='body1' >{'AC: '+this.props.Entity.ArmorClass}</Typography>
						</Grid>
						<Grid item xs={6}>
							<Typography id='HitPoints' className='body1' >{'HP: '+this.state.CurrentHitPoints+'/'+this.props.Entity.HitPoints}</Typography>
						</Grid>
						<Grid item xs={12}>
							<Typography id='Conditions' className='body1' >{conditions}</Typography>
						</Grid>
					</Grid>
				</ExpansionPanelSummary>
				<ExpansionPanelDetails>
					<Grid container spacing={8} >
						<Grid item xs={12}>
							<FormControl className='formControl' fullWidth >
								<InputLabel htmlFor="CurrentHitPoints">Current HP</InputLabel>
								<Input
									id='CurrentHitPoints'
									type='number'
									value={this.state.CurrentHitPoints != undefined ? this.state.CurrentHitPoints : ''}
									name='CurrentHitPoints'
									onChange={this.handleNumberChange} />
							</FormControl>
						</Grid>
						{EntityConditions.map(conditionName =>
							<Grid item xs={6} sm={3} md={12} lg={6} xl={4} key={conditionName}>
							<FormControlLabel
								control={
									<Switch
										id={conditionName}
										value={conditionName}
										checked={this.state[conditionName] == true}
										onChange={this.handleConditionChange}
										color="primary" />
								}
								label={conditionName}
								labelPlacement='start' />
							</Grid>
						)}
						<Grid item xs={12}>
							<Button variant="contained" value={this.props.Entity.Id} onClick={this.props.View} >
								View
							</Button>
							<Button variant="contained" value={this.props.Entity.Id} onClick={this.props.Edit} >
								Edit
							</Button>
							{/*<Button variant="contained" color="secondary" value={this.props.Entity.Id} onClick={this.deleteMonster} >
								Delete
								<DeleteIcon/>
							</Button>*/}
						</Grid>
					</Grid>
				</ExpansionPanelDetails>
			</ExpansionPanel>
		);
	}
}