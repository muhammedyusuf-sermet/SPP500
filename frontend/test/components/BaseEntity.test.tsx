import * as React from 'react';
import { mount, ReactWrapper, shallow } from 'enzyme';

import { IBaseEntityProps, IBaseEntityState, BaseEntity } from '../../src/renderer/components/platform/pages/run_encounter/entitiy/BaseEntity';

////// Happy Path //////
describe('Base Entity', () => {

	let baseEntityInstance: ReactWrapper<IBaseEntityProps, IBaseEntityState, BaseEntity>;
	const view = jest.fn();
	const edit = jest.fn();

	describe('Provided Entity values', () => {

		beforeEach(() => {
			baseEntityInstance =
				mount<BaseEntity, IBaseEntityProps, IBaseEntityState>(
					<BaseEntity
						Id={0}
						Initiative={0}
						View={view}
						Edit={edit}
						Entity={{
							EntityType: 'Monster',
							Id: 0,
							Name: 'Test Name',
							ArmorClass: 124,
							HitPoints: 623,
						}} />
				);
		})

		it('renders without crashing', () => {
			expect(baseEntityInstance).toBeDefined();
		});

		it('should have the correct string for Name', () => {
			expect(baseEntityInstance.find('Typography#Name').text()).toEqual('Test Name');
		});

		it('should have the correct number for Initiative', () => {
			expect(baseEntityInstance.find('Typography#Initiative').text()).toEqual('Initative: 0');
		});

		it('should have the correct number for ArmorClass', () => {
			expect(baseEntityInstance.find('Typography#ArmorClass').text()).toEqual('AC: 124');
		});

		it('should have the correct number for HitPoints', () => {
			expect(baseEntityInstance.find('Typography#HitPoints').text()).toEqual('HP: 623/623');
		});

		it('should have the correct string for Conditions', () => {
			expect(baseEntityInstance.find('Typography#Conditions').text()).toEqual('');
		});
	});

	describe('componentWillReceiveProps()', () => {
		beforeEach(() => {
			baseEntityInstance =
				mount<BaseEntity, IBaseEntityProps, IBaseEntityState>(
					<BaseEntity
						Id={0}
						Initiative={0}
						View={view}
						Edit={edit}
						Entity={{
							EntityType: 'Monster',
							Id: 0,
							Name: 'Test Name',
							ArmorClass: 124,
							HitPoints: 623,
						}} />
				);
		})

		it('renders without crashing', () => {
			expect(baseEntityInstance).toBeDefined();
		});

		it('renders new props', () => {
			baseEntityInstance.setProps({
				Entity: {
					EntityType: 'Monster',
					Id: 2,
					Name: 'Test Name 2',
					ArmorClass: 100,
					HitPoints: 32,
				}
			})
			expect(baseEntityInstance.find('Typography#Name').text()).toEqual('Test Name 2');
			expect(baseEntityInstance.find('Typography#ArmorClass').text()).toEqual('AC: 100');
			expect(baseEntityInstance.find('Typography#HitPoints').text()).toEqual('HP: 32/32');
			expect(baseEntityInstance.find('Typography#Conditions').text()).toEqual('');
		});

		it('renders without crashing with same props', () => {
			baseEntityInstance.setProps({
				Entity: {
					EntityType: 'Monster',
					Id: 0,
					Name: 'Test Name',
					ArmorClass: 124,
					HitPoints: 623,
				}
			})
			expect(baseEntityInstance.find('Typography#Name').text()).toEqual('Test Name');
			expect(baseEntityInstance.find('Typography#ArmorClass').text()).toEqual('AC: 124');
			expect(baseEntityInstance.find('Typography#HitPoints').text()).toEqual('HP: 623/623');
			expect(baseEntityInstance.find('Typography#Conditions').text()).toEqual('');
		});
	});

	describe('Small snapshot', () => {
		it ('matches snapshot', () => {
			const shallowBaseEntityInstance =
				shallow<BaseEntity, IBaseEntityProps, IBaseEntityState>(
					<BaseEntity
						Id={0}
						Initiative={0}
						View={view}
						Edit={edit}
						Entity={{
							EntityType: 'Monster',
							Id: 0,
							Name: 'Test Name',
							ArmorClass: 124,
							HitPoints: 623,
						}} />
				);
			expect(shallowBaseEntityInstance).toMatchSnapshot();
		});
	});

	describe('Happy Path', () => {
		// TODO: Test View and Edit buttons to make sure they call the view and edit functions.
		beforeEach(() => {
			baseEntityInstance =
				mount<BaseEntity, IBaseEntityProps, IBaseEntityState>(
					<BaseEntity
						Id={0}
						Initiative={0}
						View={view}
						Edit={edit}
						Entity={{
							EntityType: 'Monster',
							Id: 0,
							Name: 'Test Name',
							ArmorClass: 124,
							HitPoints: 623,
						}} />
				);
		})

		it('renders without crashing', () => {
			expect(baseEntityInstance).toBeDefined();
		});

		it('should be able to update state', () => {
			baseEntityInstance.find('input#CurrentHitPoints').simulate('change', { target: { value: '43' } })
			baseEntityInstance.find('input#Blinded').simulate('change')
			baseEntityInstance.find('input#Charmed').simulate('change')
			baseEntityInstance.find('input#Deafened').simulate('change')
			baseEntityInstance.find('input#Frightened').simulate('change')
			baseEntityInstance.find('input#Grappled').simulate('change')
			baseEntityInstance.find('input#Incapacitated').simulate('change')
			baseEntityInstance.find('input#Invisible').simulate('change')
			baseEntityInstance.find('input#Paralyzed').simulate('change')
			baseEntityInstance.find('input#Petrified').simulate('change')
			baseEntityInstance.find('input#Poisoned').simulate('change')
			baseEntityInstance.find('input#Prone').simulate('change')
			baseEntityInstance.find('input#Restrained').simulate('change')
			baseEntityInstance.find('input#Stunned').simulate('change')
			baseEntityInstance.find('input#Unconscious').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 43,
				Temporary: 0,
				Blinded: true,
				Charmed: true,
				Deafened: true,
				Frightened: true,
				Grappled: true,
				Incapacitated: true,
				Invisible: true,
				Paralyzed: true,
				Petrified: true,
				Poisoned: true,
				Prone: true,
				Restrained: true,
				Stunned: true,
				Unconscious: true,
			});
		});
		// STATE CHANGE
		it('should change only CurrentHitPoints when CurrentHitPoints is changed', () => {
			baseEntityInstance.find('input#CurrentHitPoints').simulate('change', { target: { value: '958' } })
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 958,
				Temporary: 0,
			});
		})

		it('should change only Blinded when Blinded is changed', () => {
			baseEntityInstance.find('input#Blinded').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Blinded: true
			});
		})

		it('should change only Charmed when Charmed is changed', () => {
			baseEntityInstance.find('input#Charmed').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Charmed: true
			});
		})

		it('should change only Deafened when Deafened is changed', () => {
			baseEntityInstance.find('input#Deafened').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Deafened: true
			});
		})

		it('should change only Frightened when Frightened is changed', () => {
			baseEntityInstance.find('input#Frightened').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Frightened: true
			});
		})

		it('should change only Grappled when Grappled is changed', () => {
			baseEntityInstance.find('input#Grappled').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Grappled: true
			});
		})

		it('should change only Incapacitated when Incapacitated is changed', () => {
			baseEntityInstance.find('input#Incapacitated').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Incapacitated: true
			});
		})

		it('should change only Invisible when Invisible is changed', () => {
			baseEntityInstance.find('input#Invisible').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Invisible: true
			});
		})

		it('should change only Paralyzed when Paralyzed is changed', () => {
			baseEntityInstance.find('input#Paralyzed').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Paralyzed: true
			});
		})

		it('should change only Petrified when Petrified is changed', () => {
			baseEntityInstance.find('input#Petrified').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Petrified: true
			});
		})

		it('should change only Poisoned when Poisoned is changed', () => {
			baseEntityInstance.find('input#Poisoned').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Poisoned: true
			});
		})

		it('should change only Prone when Prone is changed', () => {
			baseEntityInstance.find('input#Prone').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Prone: true
			});
		})

		it('should change only Restrained when Restrained is changed', () => {
			baseEntityInstance.find('input#Restrained').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Restrained: true
			});
		})

		it('should change only Stunned when Stunned is changed', () => {
			baseEntityInstance.find('input#Stunned').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Stunned: true
			});
		})

		it('should change only Unconscious when Unconscious is changed', () => {
			baseEntityInstance.find('input#Unconscious').simulate('change')
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 623,
				Temporary: 0,
				Unconscious: true
			});
		})
		// VALIDATION
		it('should change only CurrentHitPoints when CurrentHitPoints is changed', () => {
			baseEntityInstance.find('input#CurrentHitPoints').simulate('change', { target: { value: '--1' } })
			expect(baseEntityInstance.state()).toEqual({
				CurrentHitPoints: 0,
				Temporary: 0,
			});
		})
	});
});