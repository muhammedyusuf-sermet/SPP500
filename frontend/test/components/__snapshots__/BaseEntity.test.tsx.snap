// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Entity Small snapshot matches snapshot 1`] = `
<WithStyles(ExpansionPanel)
  CollapseProps={
    Object {
      "timeout": 100,
    }
  }
>
  <WithStyles(ExpansionPanelSummary)
    expandIcon={<pure(ExpandMoreIcon) />}
  >
    <WithStyles(Grid)
      container={true}
      spacing={8}
    >
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          id="Name"
          variant="h6"
        >
          Test Name
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          className="body1"
          id="Initiative"
        >
          Initative: 0
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          className="body1"
          id="ArmorClass"
        >
          AC: 124
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={6}
      >
        <WithStyles(Typography)
          className="body1"
          id="HitPoints"
        >
          HP: 623/623
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(Typography)
          className="body1"
          id="Conditions"
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(ExpansionPanelSummary)>
  <WithStyles(ExpansionPanelDetails)>
    <WithStyles(Grid)
      container={true}
      spacing={8}
    >
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(FormControl)
          className="formControl"
          fullWidth={true}
        >
          <WithStyles(WithFormControlContext(InputLabel))
            htmlFor="CurrentHitPoints"
          >
            Current HP
          </WithStyles(WithFormControlContext(InputLabel))>
          <WithStyles(Input)
            id="CurrentHitPoints"
            name="CurrentHitPoints"
            onChange={[Function]}
            type="number"
            value={623}
          />
        </WithStyles(FormControl)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(FormControl)
          className="formControl"
          fullWidth={true}
        >
          <WithStyles(WithFormControlContext(InputLabel))
            htmlFor="Temporary"
          >
            Temporary HP
          </WithStyles(WithFormControlContext(InputLabel))>
          <WithStyles(Input)
            id="Temporary"
            name="Temporary"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </WithStyles(FormControl)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Blinded"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Blinded"
              onChange={[Function]}
              value="Blinded"
            />
          }
          label="Blinded"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Charmed"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Charmed"
              onChange={[Function]}
              value="Charmed"
            />
          }
          label="Charmed"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Deafened"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Deafened"
              onChange={[Function]}
              value="Deafened"
            />
          }
          label="Deafened"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Frightened"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Frightened"
              onChange={[Function]}
              value="Frightened"
            />
          }
          label="Frightened"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Grappled"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Grappled"
              onChange={[Function]}
              value="Grappled"
            />
          }
          label="Grappled"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Incapacitated"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Incapacitated"
              onChange={[Function]}
              value="Incapacitated"
            />
          }
          label="Incapacitated"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Invisible"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Invisible"
              onChange={[Function]}
              value="Invisible"
            />
          }
          label="Invisible"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Paralyzed"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Paralyzed"
              onChange={[Function]}
              value="Paralyzed"
            />
          }
          label="Paralyzed"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Petrified"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Petrified"
              onChange={[Function]}
              value="Petrified"
            />
          }
          label="Petrified"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Poisoned"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Poisoned"
              onChange={[Function]}
              value="Poisoned"
            />
          }
          label="Poisoned"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Prone"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Prone"
              onChange={[Function]}
              value="Prone"
            />
          }
          label="Prone"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Restrained"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Restrained"
              onChange={[Function]}
              value="Restrained"
            />
          }
          label="Restrained"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Stunned"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Stunned"
              onChange={[Function]}
              value="Stunned"
            />
          }
          label="Stunned"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        key="Unconscious"
        lg={6}
        md={12}
        sm={3}
        xl={4}
        xs={6}
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Switch)
              checked={false}
              color="primary"
              id="Unconscious"
              onChange={[Function]}
              value="Unconscious"
            />
          }
          label="Unconscious"
          labelPlacement="start"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <WithStyles(Button)
          id="View0"
          onClick={[MockFunction]}
          value={0}
          variant="contained"
        >
          View
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(ExpansionPanelDetails)>
</WithStyles(ExpansionPanel)>
`;
